### DISCUSSION TITLE
Supporting Databricks UC with the Spark connector

### DISCUSSION THREAD

**Discussion by ni-mi** [unitycatalog/unitycatalog#486] [Ideas]
Currently, the Spark connector does not support Databricks UC.

Issues I currently know about:
- TemporaryTableCredentialsApi is not exposed in public Databricks UC.
- TableType supports only Managed and External- and not supported in OSS UC (VIEW | MATERIALIZED_VIEW | STREAMING_TABLE | MANAGED_SHALLOW_CLONE | FOREIGN | EXTERNAL_SHALLOW_CLONE are not supported in OSS and causes exception)
- Create table is not exposed in Databricks UC API 
- Create schema in Databricks UC API takes storage_root as a parameter - OSS UC doesn't

**Describe the solution you would like**

Until the the full API will be compatible and exposed - create properties that:

1. Will avoid using TemporaryTableCredentialsApi
2. Instead of getting exception for unsupported table types - either set their types manually to Unsupported - or - remove the unsupported types of tables from the returned results
3. Either through "UnsupportedOperationException" when trying to create a table - or - provide URL of DataWarehouse - and instead of creating a table using rest API, until it will be supported - use JDBC to do that
4. Works with no workaround

**Comment by MrPowers** [comment]
Converted this to a discussion to allow for better brainstorming!
